src/Sylius/Component/Taxonomy/Positioner/TaxonPositioner.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Taxonomy\Positioner;
+
+use Sylius\Component\Taxonomy\Model\TaxonInterface;
+use Sylius\Component\Taxonomy\Repository\TaxonRepositoryInterface;
+
+final class TaxonPositioner implements TaxonPositionerInterface
+{
+    public function __construct (
+        private TaxonRepositoryInterface $taxonRepository,
+    ) {
+    }
+
+    public function moveUp(TaxonInterface $taxon): void
+    {
+        $taxonAbove = $this->taxonRepository->findOneAbove($taxon);
```

Comment:
Is this taking one above (or below) taxon really needed here? We could only check the current position of the taxon and properly change it here without getting the one from db.

If I see correctly, the only problem that was with the earlier implementation was that when modifying the position at move up, we subtracted by making the position -1 when the taxon had 0 before and it was moved to the end. All that was really needed was to check the existing position before modifying even in JS, and that should work 🤔.